classes/controller/edit_controller.php

```diff
@@ -161,7 +161,7 @@ public function add_discussion_action() {
             $context = $PAGE->context;
             $course  = $PAGE->course;
 
-            if ($groupid == 0) {
+            if ($groupid === 0) {
```

Comment:
Are you sure that you want the strict comparison? Will it always be same type? If not an int, what happens?

Reply:
yep it will always return an int from the groups table, but updating to equals to comparison